docs/adr/adr-013-blocksync-overhaul-part-2.md

```diff
@@ -0,0 +1,535 @@
+
+The protocol is designed over plain libp2p streams with `/shrex/eds/0.0.1` as the protocol ID.
+
+#### Protobuf Schema
+
+##### Request
+
+```protobuf
+syntax = "proto3";
+
+message EDSRequest {
+  bytes hash = 1; // identifies the requested EDS.
+}
+```
+
+##### Response
+
+```protobuf
+syntax = "proto3";
+
+enum Status {
+  INVALID = 0;
+  OK = 100; // data found
+  NOT_FOUND = 200; // data not found
+  REFUSED = 201; // request refused
+}
+
+message EDSResponse {
+  Status status = 1;
+}
+```
+
+#### Streaming
+
+After flushing `EDSResponse`, a server starts streaming an ODS as a CAR file (TODO link) to a client over a libp2p 
+stream.
+
+#### Backpressure
+
+The streaming part of the protocol does not provide any mechanics for backpressure and offloads them to the layer below.
+A client-side buffer with a reasonable default size may still be added.
+
+[Backpressure][backpressure-quic] is inherited from the QUIC. The underlying default reliable transport and stream
+multiplexing protocol used in `celestia-node`.
+
+The standard TCP transport and yamux multiplexer combination do not provide reliable backpressure on the stream level,
+unlike QUIC. Therefore, TCP+yamux is not recommended and may lead to unexpected stream and connection resets in highly
+congested links.
+
+#### Flow
+
+```
+                    ┌──────┐          ┌──────┐          
+                    │Client│          │Server│          
+                    └──┬───┘          └──┬───┘          
+                       │   Open Stream   │              
+                       │ ────────────────>              
+                       │                 │              
+                       │    EDSRequest   │              
+                       │ ────────────────>              
+                       │                 │              
+                       │   EDSResponse   │              
+                       │ <────────────────              
+                       │                 │              
+                       │                 │              
+          ╔══════╤═════╪═════════════════╪═════════════╗
+          ║ ALT  │  STATUS:OK            │             ║
+          ╟──────┘     │                 │             ║
+          ║            │     CAR File    │             ║
+          ║            │ <────────────────             ║
+          ╚════════════╪═════════════════╪═════════════╝
+                       │                 │              
+                       │   Stream Close  │              
+                       │ <────────────────              
+                    ┌──┴───┐          ┌──┴───┐          
+                    │Client│          │Server│          
+                    └──────┘          └──────┘          
+
+```
+
+### Misbehavior
+
+* Server sends ODS CAR file that fails EDS reconstruction.
+* Client terminates stream after receiving the EDSResponse.
```

Comment:
Why would this be a misbehavior @Wondertan ? Can you help me understand why it matters who closes the stream ?